htdocs/societe/class/societe.class.php

```diff
@@ -3793,11 +3795,14 @@ public function create_from_member(Adherent $member, $socname = '', $socalias =
 		$this->client = 1; // A member is a customer by default
 		$this->code_client = ($customercode ? $customercode : -1);
 		$this->code_fournisseur = -1;
+		$this->typent_code = ($member->morphy == 'phy' ? 'TE_PRIVATE' : 0);
+		$this->typent_id = dol_getIdFromCode($this->db, $this->typent_code, 'c_typent', 'id', 'code');
```

Comment:
You can call the dol_getIdFromCode only if code is 'TE_PRIVATE' to avoid a useless sql access if empty.

Reply:
Done